src/components/CreateMenu/BaseCreateMenu.js

```diff
@@ -0,0 +1,86 @@
+import React, {PureComponent} from 'react';
+import {View} from 'react-native';
+import PropTypes from 'prop-types';
+import Popover from '../Popover';
+import styles from '../../styles/styles';
+import withWindowDimensions, {windowDimensionsPropTypes} from '../withWindowDimensions';
+import MenuItem from '../MenuItem';
+
+const propTypes = {
+    // Callback to fire on request to modal close
+    onClose: PropTypes.func.isRequired,
+
+    // State that determines whether to display the create menu or not
+    isVisible: PropTypes.bool.isRequired,
+
+    // Callback to fire when a CreateMenu item is selected
+    onItemSelected: PropTypes.func.isRequired,
+
+    // Menu items to be rendered on the list
+    menuItems: PropTypes.arrayOf(
+        PropTypes.shape({
+            icon: PropTypes.func.isRequired,
```

Comment:
Each of these need inline comments too

Reply:
Yeah, no prob, I'm actually not the original author, I just moved CreateMenu.js -> CreateMenu/BaseCreateMenu.js, but I'll do my best to describe these